### ISSUE TITLE
[Question]: 您好，最近在用paddlenlp的UIE模型，刚刚开始学习NLP，有几个问题想请教一下

### ISSUE THREAD

**Issue by lzh1998-jansen** [PaddlePaddle/PaddleNLP#4663]
### 请提出你的问题

我最近在用UIE，UIE是基于ernie，ernie应该也是基于bert的。bert是只用transformer里encoder结构，我很好奇，encoder架构是如何完成信息抽取中的实体抽取的的，举个例子设置的schema=["时间"，“地点”]，在预测的时候，这个shcema是起到什么作用，encoder对输入数据进行编码后，再经过全连接层，输出的数据中【cls】有语义信息，这个语义信息是怎么根据schema，进行实体抽取预测的？

**Comment by linjieccc** [comment]
@lzh1998-jansen Hi，schema可以理解为一组prompt集合，实体抽取中Encoder的输入是prompt+文本，就是匹配prompt和原文描述进行结果定位

**Comment by lzh1998-jansen** [comment]
> 

您好，能说的再详细一点吗，在预测的过程中，prompt 和待预测的文本，prompt是怎么进行处理的，我了解到文本会进行embedding操作，然后进入encoder，然后+全连接，然后预测答案的开始和结束位置。（我理解的对吗）

**Comment by linjieccc** [comment]
prompt和文本会进行拼接一起作为Encoder的输入，可以参考下面的模型结构图

![image](https://user-images.githubusercontent.com/40840292/217423241-204ec8a3-716c-40cb-b7bd-5177db1ee221.png)

**Comment by lzh1998-jansen** [comment]
![image](https://user-images.githubusercontent.com/61367424/217425238-9f1d05c9-5a09-4627-8f92-924338ad4241.png)
为什么start和end 不是在一层输出的，这样的设计有什么好处吗

**Comment by hopegithub** [comment]
criterion = paddle.nn.BCELoss()

    def uie_loss_func(outputs, labels):
        start_ids, end_ids = labels
        start_prob, end_prob = outputs
        start_ids = paddle.cast(start_ids, "float32")
        end_ids = paddle.cast(end_ids, "float32")
        loss_start = criterion(start_prob, start_ids)
        loss_end = criterion(end_prob, end_ids)
        loss = (loss_start + loss_end) / 2.0
        return loss
通过阅读代码，可以看出，UIE方法是机器阅读理解，输入是query=prompt，输出是span位置预测，预测多个span的开始和结束位置。相当于类别放在了prompt上，输出是和prompt相关的实体。
